### PR TITLE
Update to tokio 0.2 and futures 0.3

### PR DESCRIPTION
**Pull request by pimeys**

### CODE REVIEW

**Comment by pimeys** [comment]
Dum dum. Need to add the new feature flags to buildkite.

File: src/connector/postgres.rs [hunk 1]
```diff
@@ -85,12 +85,12 @@ impl SslParams {
         auth.accept_mode(self.ssl_accept_mode);
 
         if let Some(ref cert_file) = self.certificate_file {
-            let cert = fs::read(cert_file).await?;
+            let cert = fs::read(cert_file)?;
```
Comment (tomhoule) [inline src/connector/postgres.rs]: hmmm I don't know what the right thing to do would be here if we don't want to always depend on async-std, but reverting to blocking file IO feels bad

File: src/connector/connection_info.rs [hunk 2]
```diff
@@ -260,6 +226,7 @@ impl fmt::Display for SqlFamily {
 
 #[cfg(test)]
 mod tests {
+    #[cfg(feature = "sqlite")]
```
Comment (tomhoule) [inline src/connector/connection_info.rs]: why only on sqlite?

File: Cargo.toml [hunk 3]
```diff
@@ -17,11 +17,22 @@ keywords    = ["mysql", "postgresql", "sqlite", "sql"]
...
+
+full = ["pooled", "sqlite", "json-1", "postgresql", "uuid-0_7", "chrono-0_4", "mysql"]
+full-postgresql = ["pooled", "postgresql", "json-1", "uuid-0_7", "chrono-0_4", "array"]
+full-mysql = ["pooled", "mysql", "json-1", "uuid-0_7", "chrono-0_4"]
+full-sqlite = ["pooled", "sqlite", "json-1", "uuid-0_7", "chrono-0_4"]
+
+single = ["sqlite", "json-1", "postgresql", "uuid-0_7", "chrono-0_4", "mysql"]
+single-postgresql = ["postgresql", "json-1", "uuid-0_7", "chrono-0_4", "array"]
+single-mysql = ["mysql", "json-1", "uuid-0_7", "chrono-0_4"]
+single-sqlite = ["sqlite", "json-1", "uuid-0_7", "chrono-0_4"]
```
Comment (tomhoule) [inline Cargo.toml]: :100: I prefer named features, looks nice!

File: src/connector/postgres.rs (hunk 1 above)
Comment (pimeys) [reply src/connector/postgres.rs]: I don't want to rely on a runtime that spawns random threads on connect. The facilities should come from the current runtime or we just block. More we rely on certain runtime functions, harder it is to support other runtimes.

File: src/connector/connection_info.rs (hunk 2 above)
Comment (pimeys) [reply src/connector/connection_info.rs]: When you compile without sqlite feature, this gives a warning.

File: Cargo.toml (hunk 3 above)
Comment (pimeys) [reply Cargo.toml]: Yes. Some logic here finally.

**Pull request merged**